src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -376,20 +388,33 @@ static void dds_os_put_bytes_aligned (dds_ostream_t * __restrict os, const void
 static uint32_t get_type_size (enum dds_stream_typecode type)
 {
   DDSRT_STATIC_ASSERT (DDS_OP_VAL_1BY == 1 && DDS_OP_VAL_2BY == 2 && DDS_OP_VAL_4BY == 3 && DDS_OP_VAL_8BY == 4);
-  assert (type == DDS_OP_VAL_1BY || type == DDS_OP_VAL_2BY || type == DDS_OP_VAL_4BY || type == DDS_OP_VAL_8BY || type == DDS_OP_VAL_ENU);
-  if (type == DDS_OP_VAL_ENU)
-    return 4;
+  assert (type == DDS_OP_VAL_1BY || type == DDS_OP_VAL_2BY || type == DDS_OP_VAL_4BY || type == DDS_OP_VAL_8BY);
   return (uint32_t)1 << ((uint32_t) type - 1);
 }
 
+static enum enum_cdr_sz get_enum_cdr_size (uint32_t flags)
+{
+  uint32_t flag_value = ((flags & DDS_OP_FLAG_SZ2) ? 2 : 0) + ((flags & DDS_OP_FLAG_SZ1) ? 1 : 0);
```

Comment:
I think using log2(size) (see earlier) indeed makes sense, then this becomes 1 << ((flags >> something) & 0x3) (provided there is also a static assert that the two flag bits are adjacent; alternatively solved by defining, e.g., MASK and SHIFT instead of SZ2 and SZ2)